src/com/google/enterprise/adaptor/DocIdSender.java

```diff
@@ -337,16 +411,28 @@ public GroupPrincipal pushGroupDefinitions(
       double mean = ((double) numMembers) / numGroups;
       log.finer("mean size of groups: " + mean);
     }
+
+    // Remember the latest target group source fed. If we just did a
+    // full group feed, delete the previous target group source entries.
+    String oldGroupSource = groupSources.put(groupSource, feedSourceName);
+    if (feedType == FULL && (oldGroupSource != null)) {
+      // Delete any entries from the older group source by pushing an
+      // empty full feed.
+      log.log(Level.FINE, "Deleting entries from previous group source {0}",
+          oldGroupSource);
+      List<Map.Entry<GroupPrincipal, T>> emptyList = Collections.emptyList();
+      pushSizedBatchOfGroups(emptyList, caseSensitive, FULL, oldGroupSource,
```

Comment:
If I understand updated logic correctly we will have three feed files.
E.g. We have groupSource = "feed". Most Adaptors (as well as AD) have IncrementalListing and FullListing processes.
For IncrementalListing we will use incremental pushes and we will works with data source "feed". 
With FullListing we will use full pushes so we will works with two data sources "feed-A" and "feed-B". 
At the and of "full" push we will clean elder data source from two ("feed-A" or "feed-B"). But in case if some groups was added with IncrementalListing process and was removed it still will be existed in data source "feed". So these groups will never be deleted.
And the same if you launch updated adaptor without resetting index. GSA fill already contains "feed" data source that could contains groups that never be deleted.

It my thoughts are correct we need to fix this. But in fact the fix is elementary. just need to remove "feed" data source as well. I guess:
pushSizedBatchOfGroups(emptyList, caseSensitive, FULL, groupSource, handler);